WordPress/gutenberg
# Tooltip: refine existing tests

## What?


Refining the existing tooltip tests (and likely adding a few more but this is still a WIP)

## Why?


In preparation for a new tooltip component #48222, I wanted to refine and improve the existing tests for the legacy tooltip. As well as add any additional tests that are missing / would be beneficial. 

## How?


- By replacing `focus()` with `userEvent.tab()`to closer mimic user behaviour 
- Removing `act()` and fake timers 
- Replacing the button with our Button component (initially done because of [this issue](https://github.com/WordPress/gutenberg/issues/43129) but it looks like it might be related to the Icon / possibly other components. Still needs to be researched further)
- Refining the shortcut test to include what the prop is for

## Testing Instructions

Run `npm run test:unit packages/components/src/tooltip/test/index.js` and ensure tests are passing
